lib/transaction/transaction.js

```diff
@@ -828,17 +828,22 @@ Transaction.prototype._updateChangeOutput = function () {
   if (!_.isUndefined(this._changeIndex)) {
     this._removeOutput(this._changeIndex)
   }
+  this._changeIndex = this.outputs.length
+  this._addOutput(new Output({
+    script: this._changeScript,
+    satoshis: 0
+  }))
```

Comment:
what are you changing here and why?

Reply:
Adding a dummy change output with 0 value. This is used for estimating transaction size accurately, without altering the transactions unspent value.